### ISSUE TITLE
{shared,static}_library don't add deps to Libs.private in pkgconfig

### ISSUE THREAD

**Issue by guijan** [mesonbuild/meson#11023]
**Describe the bug**
In the pkgconfig module, the dependencies of a `lib` object generated with `static_library()` or `shared_library()` are not added to the .pc file's `Libs.private` field like they are with `lib` objects generated by `library()`.

**To Reproduce**
- Download and extract this minimal test project: 
[shim.zip](https://github.com/mesonbuild/meson/files/9983889/shim.zip)

- Run `meson setup build` inside the project's directory
- Compare the pkg-config files in _build/meson-private/*.pc_. Notice that _shim.pc_ has a `Libs.private` field, but _shim_shared.pc_ and _shim_static.pc_ don't.

**Expected behavior**
_shim_static.pc_ and _shim_shared.pc_ should have the missing field. Notice that `Libs.private` is for static linking only, so perhaps _shim_shared.pc_ should not have it, but consider that Meson generates the field for `lib` objects generated by `library()` even if `default_library` is set to `static`, so it might make more sense to follow the preexisting behavior of `library()` which is useful when writing Windows libraries.

**system parameters**
This bug is present on OpenBSD 7.2 with Meson 0.63.2 and Ninja 1.11.1, and also MSYS2 on Windows 10 with Meson 0.63.3 and Ninja 1.11.1. The operating system doesn't seem to matter.

**Comment by xclaesse** [comment]
@guijan by default Meson only build a shared library, so Libs.private is useless. If you setup with --default-library=both then you should see it.

**Comment by xclaesse** [comment]
TL;DR: This is intentional.

It is basically a crappy compromise because pkg-config lacks distinction between private and internal dependencies. Some details can be found in https://github.com/mesonbuild/meson/commit/5eef325ab61820ca0c3fa5673afc917dde2aa1ef and https://bugs.freedesktop.org/show_bug.cgi?id=105572.
